algorithms/kernel/dbscan/oneapi/dbscan_dense_default_batch_ucapi_impl.i

```diff
@@ -0,0 +1,511 @@
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  Implementation of default method for DBSCAN algorithm on GPU.
+//--
+*/
+
+#include "algorithms/dbscan/dbscan_types.h"
+#include "algorithms/kernel/dbscan/oneapi/dbscan_kernel_ucapi.h"
+#include "algorithms/kernel/dbscan/oneapi/cl_kernels/dbscan_cl_kernels.cl"
+#include "externals/service_ittnotify.h"
+
+using namespace daal::services;
+using namespace daal::oneapi::internal;
+
+namespace daal
+{
+namespace algorithms
+{
+namespace dbscan
+{
+namespace internal
+{
+struct QueueBlock
+{
+    QueueBlock(uint32_t queueBegin, uint32_t queueEnd, uint32_t maxQueueBlockSize)
+    {
+        count = queueEnd - queueBegin;
+        if (count > maxQueueBlockSize)
+        {
+            count = maxQueueBlockSize;
+        }
+    }
+    uint32_t count;
+};
+
+template <typename algorithmFPType>
+Status DBSCANBatchKernelUCAPI<algorithmFPType>::processResultsToCompute(DAAL_UINT64 resultsToCompute, int * const isCore, NumericTable * ntData,
+                                                                        NumericTable * ntCoreIndices, NumericTable * ntCoreObservations)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(compute.processResultsToCompute);
+    if (!isCore)
+    {
+        return Status(ErrorNullPtr);
+    }
+
+    auto & context           = Environment::getInstance()->getDefaultExecutionContext();
+    const uint32_t nRows     = ntData->getNumberOfRows();
+    const uint32_t nFeatures = ntData->getNumberOfColumns();
+
+    uint32_t nCoreObservations = 0;
+
+    for (uint32_t i = 0; i < nRows; i++)
+    {
+        if (!isCore[i])
+        {
+            continue;
+        }
+        nCoreObservations++;
+    }
+
+    if (nCoreObservations == 0)
+    {
+        return Status();
+    }
+
+    if (resultsToCompute & computeCoreIndices)
+    {
+        DAAL_CHECK_STATUS_VAR(ntCoreIndices->resize(nCoreObservations));
+        BlockDescriptor<int> indexRows;
+        DAAL_CHECK_STATUS_VAR(ntCoreIndices->getBlockOfRows(0, nCoreObservations, writeOnly, indexRows));
+        auto coreIndices = indexRows.getBuffer().toHost(ReadWriteMode::writeOnly);
+        if (!coreIndices.get())
+        {
+            return Status(ErrorNullPtr);
+        }
+
+        uint32_t pos = 0;
+        for (uint32_t i = 0; i < nRows; i++)
+        {
+            if (!isCore[i])
+            {
+                continue;
+            }
+            coreIndices.get()[pos] = i;
+            pos++;
+        }
+    }
```

Comment:
Here is no GPU-related logic. Is this code copy-pasted from CPU version of DBSCAN? If yes, I would recommend to use this piece of code directly from CPU version.